### ISSUE TITLE
SCSS code is different with single line comments inside mixins using CRLF line endings

### ISSUE THREAD

**Issue by aholtkamp** [prettier/prettier#4143]
**Prettier 1.11.1**

Using prettier on our SCSS files we experience different behaviour under linux than under windows. Further investigation revealed that the line endings make the difference. It seems when using CRLF as line endings prettier produces different output than with normal LF as line endings.

**Input (CRLF line endings):**
```scss
@mixin create-rules($padding) {
  //
  // single line comment
  //
  .abc {
    padding: $padding;
  }

  /**
   * multi line comment
   */
  .def {
    padding: $padding;
  }
}

```

**Output:**
```scss
@mixin create-rules($padding) {
  $p // single line comment
  //
  .abc {
    padding: $padding;
  }
  /* *
   * multi line comment */
  .def {
    padding: $padding;
  }
}

```

**Expected behavior:**
The file should not be changed as it already is properly formatted.

**Issues:**

- single line comments produce invalid code after formatting
- multi line comments are formatted incorrectly
- a line break between the two code blocks gets stripped

**Comment by evilebottnawi** [comment]
@aholtkamp i add tests https://github.com/prettier/prettier/pull/4145 and all works as expected, what editor you use and how you run prettier, looks something wrong with cursor

**Comment by aholtkamp** [comment]
I'm using IntelliJ IDEA. It seems your tests do not use "@mixin" but media-queries. For media-queries everything works as expected.

**Comment by evilebottnawi** [comment]
@aholtkamp add you example

**Comment by evilebottnawi** [comment]
@aholtkamp maybe you can create minimum reproducible repo or describe instruction how you run prettier?

**Comment by aholtkamp** [comment]
I will see if I can provide a repo, probably in the next few hours

**Comment by aholtkamp** [comment]
Oh, I'm sorry, it seems I have changed and formatted my code too often when trying to provide a minimal example for the issue (our affected file is much bigger).

The problem occurs when the mixin parameter list too long and will be split into multiple lines.

**Input (CRLF line endings):**
```scss
@mixin create-rules(
  $padding,
  $param2: 30px,
  $param3: 1px solid black,
  $param4: 1,
  $param5: 1
) {
  //
  // single line comment
  //
  .abc {
    padding: $padding;
  }

  /**
   * multi line comment
   */
  .def {
    padding: $padding;
  }
}

```

I hope you can reproduce it with this code, otherwise I will provide a repo.

**Comment by aholtkamp** [comment]
In addition to this the problem also occurs when the parameter list is rewritten:

**Input (CRLF line endings):**
```scss
@mixin create-rules(
  $padding
) {
  //
  // single line comment
  //
  .abc {
    padding: $padding;
  }

  /**
   * multi line comment
   */
  .def {
    padding: $padding;
  }
}

```

Of course the expected output file would be:

**Output:**
```scss
@mixin create-rules($padding) {
  //
  // single line comment
  //
  .abc {
    padding: $padding;
  }

  /**
   * multi line comment
   */
  .def {
    padding: $padding;
  }
}

```

**Comment by darekkay** [comment]
I can reproduce the issue in my project for `@mixin` and `@function` declarations and only on Windows (CRLF). On Linux/Mac there are no issues (LF).

**Comment by justrhysism** [comment]
This issue isn't only for comments, have noticed that Prettier (SCSS) has all sorts of trouble with CRLF. Worth also noting that I haven't experienced this with JS, TS or JSON using the same config.

**Input (CRLF)** - _Expected output should be unchanged_

```scss
/// Greys
$colour-map-grey: (
  40: #58606e,
  60: #8e99ab,
  80: #c8d1e0,
  100: #f5f7fa,
  default: #8e99ab
) !default;

/// Blues
$colour-map-blue: (
  30: #114599,
  50: #3272d9,
  70: #84b1fa,
  100: #f0f5fc,
  default: #3272d9
) !default;

$colour-map-green: (
  20: #1a5130,
  30: #2a844e,
  50: #57c182,
  90: #e1faeb,
  default: #2a844e
) !default;
```

**Output**
```scss
/// Greys
$colour-map-grey: (
  40: #58606e,
  60: #8e99ab,
  80: #c8d1e0,
  100: #f5f7fa,
  default: #8e99ab
) !default;

// / Blues
$colour-map-blue: (
  30: #114599,
  50: #3272d9,
  70: #84b1fa,
  100: #f0f5fc,
  default: #3272d9
) !default;
$colour-map-green: (
  20: #1a5130,
  30: #2a844e,
  50: #57c182,
  90: #e1faeb,
  default: #2a844e
) !default;
```

**Comment by thorn0** [comment]
Fixed at some point between versions 1.15 and 1.16.
